pages/blueprintEdit/index.js

```diff
@@ -644,13 +644,15 @@ class EditBlueprintPage extends React.Component {
                   />
                 </div>
               )}
-            <ComponentInputs
-              label={formatMessage(messages.listTitleAvailableComps)}
-              components={inputComponents[inputs.selectedInputPage]}
-              handleComponentDetails={this.handleComponentDetails}
-              handleAddComponent={this.handleAddComponent}
-              handleRemoveComponent={this.handleRemoveComponent}
-            />
+            {(blueprint.components !== undefined && blueprint.packages !== undefined) &&
```

Comment:
I'm wondering if this condition should be moved [higher](https://github.com/osbuild/cockpit-composer/pull/1018/files#diff-268ff6cc6477c8bc85a815bc0b236684R574) and combined with the following condition:
`{(inputComponents !== undefined && (`

Because including this condition here results in the following result:
![image](https://user-images.githubusercontent.com/21063328/85782594-ac775a80-b6f4-11ea-91ef-f0f6214d2e60.png)

But if we move it higher, I think we'd continue to see the loading state while we wait for the blueprint to finish loading, which seems like a better behavior:
<img width="516" alt="image" src="https://user-images.githubusercontent.com/21063328/85796396-e3a33700-b707-11ea-87ae-fafaf8297119.png">

BTW, I can't actually test this condition, because the blueprint always loads faster than the inputs, and I'm not sure how to reproduce the reverse.

Reply:
Thanks for the suggestion! I moved the conditional up. I can also not reproduce it. This is just my best attempt to fix #1005 without being able to replicate it.